nck/utils/file_reader.py

```diff
@@ -53,20 +52,42 @@ def format_csv_fieldnames(csv_fieldnames):
     elif isinstance(csv_fieldnames, (str, bytes)):
         _csv_fieldnames = json.loads(csv_fieldnames)
     else:
-        raise TypeError(
-            f"The CSV fieldnames is of the following type: {type(csv_fieldnames)}."
-        )
+        raise TypeError(f"The CSV fieldnames is of the following type: {type(csv_fieldnames)}.")
     assert isinstance(_csv_fieldnames, list)
     return _csv_fieldnames
 
 
-class CSVReader(object):
+class FormatReader:
```

Comment:
Do we really need a class here. Maybe we could keep things simple with just the `create_file_reader` function. What do you think?

Reply:
yes you are totally right.